File: backend/tiggle/src/main/java/com/side/tiggle/domain/txtag/TxTagDto.java
package com.side.tiggle.domain.txtag;

import com.side.tiggle.domain.txtag.model.TxTag;
import lombok.*;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder
public class TxTagDto {
    long id;
    long txId;
    long memberId;
    String tagNames;

    public static TxTagDto fromEntity(TxTag txTag) {
        return TxTagDto.builder()
                .id(txTag.getId())
                .txId(txTag.getTxId())
                .memberId(txTag.getMemberId())
                .tagNames(txTag.getTagNames())
                .build();
    }

    public TxTag toEntity(TxTagDto dto) {
        return TxTag.builder()
                .txId(dto.getTxId())
                .memberId(dto.getMemberId())
                .tagNames(dto.getTagNames())
                .build();
    }
}

File: backend/tiggle-root/tiggle/src/main/kotlin/com/side/tiggle/domain/txtag/dto/TxTagDto.kt
package com.side.tiggle.domain.txtag.dto

import com.side.tiggle.domain.txtag.model.TxTag

data class TxTagDto (
    val id: Long,
    val txId: Long,
    val memberId: Long,
    val tagNames: String?
) {

    fun toEntity(dto: TxTagDto): TxTag {
        return TxTag(
            txId = dto.txId,
            memberId = dto.memberId,
            tagNames = dto.tagNames
        )
    }

    companion object {
        fun fromEntity(txTag: TxTag): TxTagDto {
            return TxTagDto(
                id = txTag.id,
                txId = txTag.txId,
                memberId = txTag.memberId,
                tagNames = txTag.tagNames
            )
        }
    }
}